dataservice/api/study/resources.py

```diff
@@ -0,0 +1,134 @@
+from flask import abort, request
+from marshmallow import ValidationError
+
+from dataservice.extensions import db
+from dataservice.api.common.pagination import paginated, Pagination
+from dataservice.api.study.models import Study
+from dataservice.api.study.schemas import StudySchema
+from dataservice.api.common.views import CRUDView
+
+
+class StudyListAPI(CRUDView):
+    """
+    Study API
+    """
+    endpoint = 'studies_list'
+    rule = '/studies'
+    schemas = {'Study': StudySchema}
+
+    @paginated
+    def get(self, after, limit):
+        """
+        Get a paginated studies
+        ---
+        template:
+          path:
+            get_list.yml
+          properties:
+            resource:
+              Study
+        """
+        q = Study.query
+
+        return (StudySchema(many=True)
+                .jsonify(Pagination(q, after, limit)))
+
+    def post(self):
+        """
+        Create a new study
+        ---
+        template:
+          path:
+            new_resource.yml
+          properties:
+            resource:
+              Study
+        """
+        try:
+            st = StudySchema(strict=True).load(request.json).data
+        except ValidationError as err:
+            abort(400, 'could not create study: {}'.format(err.messages))
+
+        db.session.add(st)
+        db.session.commit()
+        return StudySchema(
+            201, 'study {} created'.format(st.kf_id)
+        ).jsonify(st), 201
+
+
+class StudyAPI(CRUDView):
+    """
+    Study API
+    """
+    endpoint = 'studies'
+    rule = '/studies/<string:kf_id>'
+    schemas = {'Study': StudySchema}
+
+    def get(self, kf_id):
+        """
+        Get a study by id
+        ---
+        template:
+          path:
+            get_by_id.yml
+          properties:
+            resource:
+              Study
+        """
+        st = Study.query.get(kf_id)
+        if st is None:
+            abort(404, 'could not find {} `{}`'
+                  .format('study', kf_id))
+        return StudySchema().jsonify(st)
+
+    def patch(self, kf_id):
+        """
+        Update fields in a study and preserve existing fields.
+        ---
+        template:
+          path:
+            update_by_id.yml
+          properties:
+            resource:
+              Study
+        """
+        body = request.json
+        st = Study.query.get(kf_id)
+        if st is None:
+            abort(404, 'could not find {} `{}`'
+                  .format('study', kf_id))
+
+        try:
+            st = (StudySchema(strict=True).load(body, instance=st,
+                                                partial=True).data)
+        except ValidationError as err:
+            abort(400, 'could not update participant: {}'.format(err.messages))
```

Comment:
Prob copy/paste error, change `could not update participant` to `could not update study`